pkg/kubelet/status/status_manager.go

```diff
@@ -918,10 +918,13 @@ func GetSidecarsStatus(pod *v1.Pod) SidecarsStatus {
 					} else {
 						klog.Infof("Pod %s: %s: sidecar is ready", format.Pod(pod), container.Name)
 					}
-				} else if status.State.Waiting != nil {
-					// check if non-sidecars have started
-					klog.Infof("Pod: %s: %s: non-sidecar waiting", format.Pod(pod), container.Name)
-					sidecarsStatus.ContainersWaiting = true
+				} else {
+					if (status.State.Waiting == nil && status.State.Running == nil && status.State.Terminated == nil) ||
```

Comment:
State is a value object and not a pointer, so to check for empty/unknown state kinda have to check all fields are nil.. feels clunky and might break if there are more fields.

Reply:
In the all nil case we can compare the zero value - this can still potentially break if new fields are added, but maybe this is a bit safer:
```suggestion
					if (status.State == v1.ContainerState{}) || status.State.Waiting != nil
```

https://go.dev/play/p/KmoJ9g__5pp

Reply:
changed to comparing to zero value.